src/chart/helper/EffectLine.js

```diff
@@ -187,7 +188,27 @@ effectLineProto.updateSymbolPosition = function (symbol) {
     var ty = quadraticDerivativeAt(p1[1], cp1[1], p2[1], t);
 
     symbol.rotation = -Math.atan2(ty, tx) - Math.PI / 2;
-
+    // enable continuity trail for 'line', 'rect', 'roundRect' symbolType
+    if (this._symbolType === 'line' || this._symbolType === 'rect' || this._symbolType === 'roundRect') {
+        if (
+            lastPos[0] !== 0 && lastPos[1] !== 0
+            && lastPos[0] !== p2[0] && lastPos[1] !== p2[1]
```

Comment:
I guess the logic here is to make sure `lastPos` and pos is in the same period of animation.
Perhaps it's more robust to check current `t` and previous `t`, make sure current `t` is larger than the previous.

Reply:
that's true, thanks pissang ~